components/uniset/src/uniset.rs

```diff
@@ -24,20 +25,20 @@ const BMP_MAX: u32 = 0xFFFF;
 /// and predefined ranges.
 #[derive(Debug, PartialEq, Hash, Eq, Clone, Yokeable, ZeroCopyFrom)]
 #[yoke(cloning_zcf)]
-pub struct UnicodeSet {
+pub struct UnicodeSet<'d> {
     // TODO: need advice - how should we remove Hash and Eq from UnicodeSet unless we need it?
 
     // If we wanted to use an array to keep the memory on the stack, there is an unsafe nightly feature
     // https://doc.rust-lang.org/nightly/core/array/trait.FixedSizeArray.html
     // Allows for traits of fixed size arrays
 
     // Implements an [inversion list.](https://en.wikipedia.org/wiki/Inversion_list)
-    inv_list: Vec<u32>,
+    inv_list: ZeroVec<'d, u32>,
```

Comment:
How do I solve the following compiler error(s)?:
```
error[E0277]: the trait bound `ZeroVec<'_, u32>: Hash` is not satisfied
   --> components/uniset/src/uniset.rs:36:5
    |
36  |     inv_list: ZeroVec<'d, u32>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Hash` is not implemented for `ZeroVec<'_, u32>`
    | 
   ::: /home/elango/.rustup/toolchains/1.51.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/hash/mod.rs:168:13
    |
168 |     fn hash<H: Hasher>(&self, state: &mut H);
    |             - required by this bound in `core::hash::Hash::hash`
    |
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ZeroVec<'d, u32>: Eq` is not satisfied
   --> components/uniset/src/uniset.rs:36:5
    |
36  |     inv_list: ZeroVec<'d, u32>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Eq` is not implemented for `ZeroVec<'d, u32>`
    | 
   ::: /home/elango/.rustup/toolchains/1.51.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:297:31
```

Reply:
**Hash:**

Easiest would be to stop implementing Hash on UnicodeSet.  I remember that was only used by some test code or something.  It never really made sense to me.

Otherwise, you could look into adding a Hash impl on ZeroVec.  It's probably not too hard.

**Eq:**

Seems like we need an Eq impl on ZeroVec.  You could make a separate PR to add it.